claims_p2.go

```diff
@@ -161,7 +202,16 @@ func (c P2Claims) ToCBOR() ([]byte, error) { //nolint:gocritic
 }
 
 func (c P2Claims) ToUnvalidatedCBOR() ([]byte, error) { //nolint:gocritic
+	var scs ISwComponents
+	if c.SwComponents != nil && c.SwComponents.IsEmpty() {
```

Comment:
Just checking : This is Unvalidated CBOR hence we are allowing SW Component to be Empty, right ? Otherwise, this should fail the Validate() check on 196 for the equivalent method?

Reply:
Yes, this just ensure correct marhalling, but  does not validate.